joomla/joomla-cms
# [4.0][Admin menu] Improving various aspects including multilingual

### Summary of Changes
**What this does:**
Adding badge to menus containing a home. 
Not flags as some languages may chose to not use flags.
For this, modified `sql select` in joomla, menus and modern presets 
Allow custom admin menu items to display an image or a badge (modified the xmls).
Take off links for separator menu items and make sure they are aligned correctly also for custom admin menu.

**What it does not do:**
Does not solve creating a full custom admin menu with a preset.
Does not solve some issues with custom admin menu items of type heading and container, etc.
Does not solve various issues with the "modern" preset (dropdowns, etc.)

Left in the existing presets the code `ELSE 'image:mod_languages/icon-16-language.png' END` which I think can be deleted. 
Left the image class alone for now.
Not sure what to do with Title Attribute and Link Class (EDIT: modified, see below). Left them for now.
_Waiting for advice on these._ 

### Testing Instructions
Create a multilingual site with the multilingual sample data.
Create a custom admin menu and add a few menu items.

When creating these, you will get this display, here for a Languages menu item where both icon class and image have been entered:

<img width="1425" alt="Screen Shot 2019-08-18 at 18 27 59" src="https://user-images.githubusercontent.com/869724/63227374-f98dc480-c1e5-11e9-8b5f-f9061a3a173b.png">

_The Description of the new icon link field needs certainly love and care. Maybe use `class` and not `code`_

(EDIT: modified, see below)

### After patch
For menu items containing homes:

<img width="322" alt="Screen Shot 2019-08-18 at 18 29 55" src="https://user-images.githubusercontent.com/869724/63227392-41145080-c1e6-11e9-9216-180eb51c4e2e.png">

For a custom admin menu item (Global Config has an image as icon, Container has icon but does not work

<img width="303" alt="Screen Shot 2019-08-18 at 18 40 11" src="https://user-images.githubusercontent.com/869724/63227503-aae12a00-c1e7-11e9-91fd-47563cf15394.png">

Should also be fine in RTL
